app/controllers/api/v1/puzzles_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Api::V1::PuzzlesController < ApplicationController
+
+  def index
+
+      zip_code = params[:zip_code]
+
+      users = User.where(zip_code: zip_code) if zip_code.present?
+      
+      if users != []
+        puzzles = Puzzle.where(user_id: users.pluck(:id))
+        render json: PuzzleSerializer.new(puzzles)
+      elsif puzzles == [] || users = []
+        render json: { error: "Puzzles not found in this area" }, status: "404"
```

Comment:
This error handling works for now to meet MVP within our time limitation. But in a refactor we'll want to get this working through the ErrorSerializer. : )